fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/typing/Type.java

```diff
@@ -1359,6 +1379,8 @@ class Array implements Type {
         @Nullable
         private final Type elementType;
 
+        private final boolean needsWrapper;
```

Comment:
I'm not quite sure whether `isNullable` and `needsWrapper` should be the same.

Reply:
Maybe you just have `needsWrapper()` to just do `return isNullable;`.